dynamodb/src/main/scala/uk/gov/nationalarchives/DADynamoDBClient.scala

```diff
@@ -193,6 +205,34 @@ class DADynamoDBClient[F[_]: Async](dynamoDBClient: DynamoDbAsyncClient):
       .liftF
       .flatMap(res => validateAndConvertAttributeValuesList(res.items()))
 
+  private def writeOrDeleteItems[T](
```

Comment:
Should we limit the upper bound of T to `Product` ( [T <: Product])?

Reply:
As above